// ng2-bootstrap-grid/dist/src/columns.directive.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var ColSmDirective = (function () {
    function ColSmDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.sm = 0;
    }
    ColSmDirective.prototype.ngOnInit = function () {
        var colClass = 'col';
        if (this.sm > 0)
            colClass += '-sm-' + this.sm;
        this.renderer.setElementClass(this.element.nativeElement, colClass, true);
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Number)
    ], ColSmDirective.prototype, "sm", void 0);
    ColSmDirective = __decorate([
        core_1.Directive({
            selector: 'column[sm]'
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef, core_1.Renderer])
    ], ColSmDirective);
    return ColSmDirective;
}());
exports.ColSmDirective = ColSmDirective;
var ColMdDirective = (function () {
    function ColMdDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.md = 0;
    }
    ColMdDirective.prototype.ngOnInit = function () {
        var colClass = 'col';
        if (this.md > 0)
            colClass += '-md-' + this.md;
        this.renderer.setElementClass(this.element.nativeElement, colClass, true);
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Number)
    ], ColMdDirective.prototype, "md", void 0);
    ColMdDirective = __decorate([
        core_1.Directive({
            selector: 'column[md]'
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef, core_1.Renderer])
    ], ColMdDirective);
    return ColMdDirective;
}());
exports.ColMdDirective = ColMdDirective;
var ColLgDirective = (function () {
    function ColLgDirective(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.lg = 0;
    }
    ColLgDirective.prototype.ngOnInit = function () {
        var colClass = 'col';
        if (this.lg > 0)
            colClass += '-lg-' + this.lg;
        this.renderer.setElementClass(this.element.nativeElement, colClass, true);
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Number)
    ], ColLgDirective.prototype, "lg", void 0);
    ColLgDirective = __decorate([
        core_1.Directive({
            selector: 'column[lg]'
        }), 
        __metadata('design:paramtypes', [core_1.ElementRef, core_1.Renderer])
    ], ColLgDirective);
    return ColLgDirective;
}());
exports.ColLgDirective = ColLgDirective;
